app/views.py

```diff
@@ -0,0 +1,231 @@
+from django.http import HttpResponse
+from django.shortcuts import redirect
+from .models import Board, Card, BoardList
+from django.contrib.auth.models import User
+from django.http import HttpResponseRedirect
+from django.views.generic import TemplateView
+from .forms import BoardForm, ListForm, CardForm, UserForm
+from django.contrib.auth import authenticate, login
+from django.shortcuts import render, get_object_or_404
+
+
+# Create user view class
+class LoginView(TemplateView):
+    """
+    Let user to login
+    """
+
+    template_name = 'app/login.html'
+    form = UserForm
+    def get(self, *args, **kwargs):
+        form = self.form()
+        return render(self.request, self.template_name,{ 'form' : form})
+
+    def post(self, *args, **kwargs):
+        username = self.request.POST['username']
+        password = self.request.POST['password']
+        user = authenticate(self.request, username=username, password=password)
+        if user is not None:
+            login(self.request, user)
+            return HttpResponseRedirect('/')
+        #  else:
+        # # Return an 'invalid login' error message. (to be continued)
+
+
+class Boards(TemplateView):
+    """
+    View the board/s of the user
+    """
+
+    
+    template_name = 'app/index.html'
+    def get(self, *args, **kwargs):
+        # select current user and activation
+        boards = Board.objects.filter(user=self.request.user, activation=True)
+        return render(self.request, self.template_name, {'boards' : boards})
+
+
+class AddBoard(TemplateView):
+    """
+    Let the user add new board
+    """
+
+    form = BoardForm
+    template_name = 'app/create_board.html'
+    def get(self,request):
+        form = self.form()
+        return render(request, self.template_name, {'form': form})
+
+    def post(self, request):
+        # create a form instance and populate it with data from the request:
+        form = self.form(request.POST)
+        # check whether it's valid:
+        if form.is_valid():
+            # if it is valid, save(commit=False) gets you a model object, then you can add your extra data
+            form = form.save(commit=False)
+            # add user data then save() it
+            form.user = request.user
+            form.save()
+            # redirect to a home url:
+            return HttpResponseRedirect('/')
+        return render(request, self.template_name, {'form': form})
+
+class BoardView(TemplateView):
+    """
+    Redirect the user to the selected board
+    """
+
+    template_name = 'app/board_view.html'
+    def get(self, *args, **kwargs):
+        board_id = kwargs.get('id')
+        # iterate board model, if data doesnt exist it will move you to 404 page
+        board = get_object_or_404(Board, id=board_id)
+        # iterate list model
+        board_list = BoardList.objects.filter(board=board_id)
+        # select list id
+        card_query = Card.objects.all()
```

Comment:
this is wrong, filter cards by boardList

Reply:
Done sir